components/sendgrid/sendgrid.app.js

```diff
@@ -77,39 +86,25 @@ module.exports = {
         }
       }, retryOpts);
     },
-    async searchContacts(query) {
-      const url = this._contactsSearchUrl();
-      const searchParams = {
-        url,
-        query,
-      };
-      return this._withRetries(
-        () => this._getAllItems(searchParams),
-      );
-    },
     async getWebhookSettings() {
       const url = this._webhookSettingsUrl();
       const requestConfig = this._makeRequestConfig();
-      return this._withRetries(
-        () => axios.get(url, requestConfig),
-      );
+      return this._withRetries(() => axios.get(url, requestConfig));
```

Comment:
I would also go with the only function that can deal with `axios` calls, in this case it might be called `_makeRequest(customConfig)` in that way the `axios` lib is not being called in other functions like this one but abstracted.

Reply:
hey @jcortes regarding `axios`... i checked the code base, and `axios` lib is used on sources, which are already  published and were authored by @jverce 
this pr is about sendgrid actions, not source. I think we shouldn't modify this code. don't you think?
thoughts? @dylburger @dannyroosevelt

Reply:
As per this [Slack dicussion](https://pipedream-users.slack.com/archives/G01J50YGJGP/p1645187574309249)
this and other changes noted in https://github.com/PipedreamHQ/pipedream/pull/1288#issuecomment-1049088143, I won't be addressing them because they pertain to Sendgrid sources.